### PR TITLE
Add cpp lint action

### PR DESCRIPTION
**Pull request by Teddy-1000**

### CODE REVIEW

**Comment by johtoblan** [comment]
Does not seem to do anything?   From last part of https://github.com/EURODEO/e-soh-event-queue/actions/runs/6379624677/job/17312446949 run cpp-linter

INFO:CPP Linter:Fetching files list from url: https://api.github.com/repos/EURODEO/e-soh-event-queue/pulls/26
  INFO:CPP Linter:No source files need checking!

And "fail fast?" is not run

File: .github/workflows/cpp-syntax.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.14
+      with:
+        source: './src/esoh/ingest/bufr'
+        extensions: 'h,cpp'
+        clangFormatVersion: 14
+        style: GNU
```
Comment (johtoblan) [inline .github/workflows/cpp-syntax.yml]: What are our options here?

**Comment by istvans-mn** [comment]
Created a PR #34

**Review by johtoblan — APPROVED** [review]
LGTM

**Pull request merged**